ibis/expr/api.py

```diff
@@ -4193,22 +4193,47 @@ def mutate(table, exprs=None, **mutations):
         for name, expr in sorted(mutations.items(), key=operator.itemgetter(0))
     )
 
+    # The below logic computes the mutation node exprs by splitting the
+    # assignment exprs into two disjoint sets:
+    # 1) overwriting_cols_to_expr, which maps a column name to its expr
+    # if the expr contains a column that overwrites an existing table column
+    # 2) non_overwriting_exprs, which is a list of all exprs that do not do
+    # any overwriting.
+    # Given these two data structures, we can compute the mutation node exprs
+    # based on whether any columns are being overwritten.
+    overwriting_cols_to_expr = {}
+    non_overwriting_exprs = []
+    table_schema = table.schema()
     for expr in exprs:
-        if expr.get_name() and isinstance(expr, ir.DestructColumn):
-            raise com.ExpressionError(
-                f"Cannot name a destruct column: {expr.get_name()}"
-            )
+        expr_contains_overwrite = False
+        if isinstance(expr, ir.DestructColumn):
+            if expr.get_name():
+                raise com.ExpressionError(
+                    f"Cannot name a destruct column: {expr.get_name()}"
+                )
+            for name in expr.type().names:
+                if name in table_schema:
```

Comment:
Actually there might be a bug here -

If the destruct column overwrites multi columns, you only have the first column being overwritten in the `overwriting_cols_to_expr`, and when you iterate later, I think you get the wrong results when hitting the second overwritten column. I think what you can do here is to track all columns that was overwritten, and if a overwritten column is 

(1) the first overridden column in the destruct column, put a mapping between this column and the destruct column. This effectively decides the position of the destruct column in the result
(2) the first overridden column in the destruct column, put a mapping between this column and a place holder (maybe None value) and when you hit this column in the for loop below, do nothing (since you have already added the destruct column to the list at this point, which contains all the overridden columns)

Reply:
Probably want a test for this as well

Reply:
Yes good catch, the prior implementation would error out because it essentially assumes a max of one overwritten col per DestructColumn. I believe the solution you suggested works and added a new test test_elementwise_udf_multiple_overwrite_destruct as well.